infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/fsm/ReplicatingState.java

```diff
@@ -52,6 +52,12 @@ public LogReplicationRuntimeState processEvent(LogReplicationRuntimeEvent event)
                 log.debug("Connection lost to non-leader node {}", endpointDown);
                 // If a non-leader node loses connectivity, reconnect async and continue.
                 return null;
+            case REMOTE_LEADER_LOSS:
+                if (fsm.getRemoteLeader().isPresent() && fsm.getRemoteLeader().get().equals(event.getEndpoint())) {
```

Comment:
We might want to check if there are fsm.getConnectedEndpoints? because if we are not connected to any remote endpoint we might directly jump to WaitForConnections as there is no-one to query for leadership.

Reply:
Ok